cow_amm/cow_amm_lp_and_tvl_4047078.sql

```diff
@@ -0,0 +1,151 @@
+--  {{token_b}} - other token of the desired uni pool
+--  {{start}} - date as of which the analysis should run
+
+-- Given that we might not have records every day in the source data (e.g. not every day the lp supply may change), 
+-- but still want to visualize development on a per day basis,  we create an auxiliary table with one record per 
+-- day between `start` and `now`
+with date_range as (
+    select t.day
+    from
+        unnest(sequence(
+            date(timestamp '{{start}}'),
+            date(now())
+        )) t (day) --noqa: AL01
+),
+
+-- Finds the CoW AMM pool address given tokens specified in query parameters (regardless of order) 
+cow_amm_pool as (
+    select
+        blockchain,
+        address,
+        token_1_address,
+        token_2_address
+    from query_3959044
+    where ((token_1_address = {{token_a}} and token_2_address = {{token_b}}) or (token_2_address = {{token_a}} and token_1_address = {{token_b}}))
+),
+
+-- per day lp token total supply changes of the CoW AMM pool by looking at burn/mint events
+lp_balance_delta as (
+    select
+        date(evt_block_time) as "day",
+        sum(case when "from" = 0x0000000000000000000000000000000000000000 then value else -value end) as lp_supply
+    from erc20_ethereum.evt_transfer
+    where
+        contract_address in (select address from cow_amm_pool)
+        and ("from" = 0x0000000000000000000000000000000000000000 or "to" = 0x0000000000000000000000000000000000000000)
+    group by 1
+),
+
+total_lp as (
+    select
+        day,
+        lp_supply,
+        sum(lp_supply) over (order by day) as total_lp
+    from lp_balance_delta
+),
+
+-- lp token total supply without date gaps
+lp_total_supply as (
+    select
+        day,
+        total_lp
+    from (
+        -- join full date range with potentially incomplete data. This results in many rows per day (all total supplies on or before that day)
+        -- rank() is then used to order join candidates by recency (rank = 1 is the latest lp supply)
+        select
+            date_range.day,
+            total_lp,
+            rank() over (partition by (date_range.day) order by lp.day desc) as latest
+        from date_range
+        inner join total_lp as lp
+            on date_range.day >= lp.day
+            -- performance optimisation: this assumes one week prior to start there was at least one lp supply change event
+            and lp.day >= (timestamp '{{start}}' - interval '7' day)
+    )
+    where latest = 1
+),
+
+-- Compute tvl by multiplying each day's closing price with net transfers in and out of the pool
+get_tvl as (
+    select
+        x.day,
+        sum(amount * price_close / pow(10, decimals)) as total_tvl
+    from (
+        select
+            date(evt_block_time) as "day",
+            contract_address,
+            -value as amount
+        from erc20_ethereum.evt_transfer
+        where
+            "from" in (select address from cow_amm_pool)
```

Comment:
Shouldn't we filter on the token here? Otherwise sending a random token to the USDC/WETH pool would increase its TVL which seems wrong. You can see that this did happen: https://etherscan.io/address/0xb4e16d0168e52d35cacd2c6185b44281ec28c9dc#code

![image](https://github.com/user-attachments/assets/2c370898-9db5-4801-9f57-e4b0aea69ce9)